services/app-api/handlers/reports/delete.ts

```diff
@@ -0,0 +1,32 @@
+import handler from "../handler-lib";
+import dynamoDb from "../../utils/dynamo/dynamodb-lib";
+import { hasPermissions } from "../../utils/auth/authorization";
+import {
+  NO_KEY_ERROR_MESSAGE,
+  UNAUTHORIZED_MESSAGE,
+} from "../../utils/constants/constants";
+import { StatusCodes, UserRoles } from "../../utils/types/types";
+
+export const deleteReport = handler(async (event, _context) => {
+  if (!hasPermissions(event, [UserRoles.ADMIN])) {
+    return {
+      status: StatusCodes.UNAUTHORIZED,
+      body: UNAUTHORIZED_MESSAGE,
+    };
+  } else if (
+    !event?.pathParameters?.state! ||
+    !event?.pathParameters?.reportId!
+  ) {
+    throw new Error(NO_KEY_ERROR_MESSAGE);
+  } else {
+    const params = {
+      TableName: process.env.REPORT_TABLE_NAME!,
+      Key: {
+        state: event?.pathParameters?.state!,
+        reportId: event?.pathParameters?.reportId!,
+      },
+    };
+    await dynamoDb.delete(params);
```

Comment:
Does this need to clean up the related data? Shouldn't it make a second dynamoDb.delete() for report data?

Reply:
I went back and forth on this one, but I think you're right that it makes more sense to keep that logic tied together in the backend. Changed. 👍